uportal-war/src/main/java/org/jasig/portal/portlets/portletadmin/PortletAdministrationHelper.java

```diff
@@ -253,13 +209,71 @@ public PortletDefinitionForm getPortletDefinitionForm(IPerson person, String por
         return form;
     }
 
+    /*
+     * Add groups and permissions to form.
+     */
+    private void getGroupPermissions(IPortletDefinition def, PortletDefinitionForm form, String activity) {
+        final AuthorizationService authService = AuthorizationService.instance();
+        final String portletTargetId = PermissionHelper.permissionTargetIdForPortletDefinition(def);
+
+        /* We are concerned with PORTAL_SUBSCRIBE system */
+        final IPermissionManager pm = authService.newPermissionManager(IPermission.PORTAL_SUBSCRIBE);
+
+        /* Obtain the principals that have permission for the activity on this portlet */
+        final IAuthorizationPrincipal[] principals = pm.getAuthorizedPrincipals(activity, portletTargetId) ;
+        for (IAuthorizationPrincipal principal : principals) {
+            JsonEntityBean bean;
+
+            // first assume this is a group
+            IEntityGroup group = GroupService.findGroup(principal.getKey());
+            if (group != null) {
+                bean = new JsonEntityBean(group, EntityEnum.GROUP);
+            }
+
+            // if a matching group can't be found, try to find a matching
+            // non-group entity
+            else {
+                IGroupMember member = authService.getGroupMember(principal);
+                bean = new JsonEntityBean(member, EntityEnum.PERSON);
+                String name = groupListHelper.lookupEntityName(bean);
+                bean.setName(name);
+            }
+
+            /* Make sure we capture the principal */
+            if (!form.getGroups().contains(bean)) {
+                form.addGroup(bean);
+            }
+
+            /* Assumption: group names are unique in this context */
+            form.addPermission(bean.getName() + "_" + activity);
+        }
+    }
+
+    /*
+     * Create a {@code PortletDefinitionForm} and pre-populate it with default categories and group permissions.
+     */
+    private PortletDefinitionForm createNewPortletDefinitionForm() {
+        PortletDefinitionForm form = new PortletDefinitionForm();
+
+        // pre-populate with top-level category
+        final IEntityGroup portletCategoriesGroup = GroupService.getDistinguishedGroup(GroupService.PORTLET_CATEGORIES);
+        form.addCategory(new JsonEntityBean(portletCategoriesGroup, groupListHelper.getEntityType(portletCategoriesGroup)));
+
+        // pre-populate with top-level group
+        final IEntityGroup everyoneGroup = GroupService.getDistinguishedGroup(GroupService.EVERYONE);
+        form.addGroup(new JsonEntityBean(everyoneGroup, groupListHelper.getEntityType(everyoneGroup)));
+        form.addPermission(everyoneGroup.getName() + "_" + IPermission.PORTLET_SUBSCRIBER_ACTIVITY);
```

Comment:
The group collection drives the HTML table in the form while the permissions are referenced only to set the checkbox state. The permissions list alone could be used but at the cost of more parsing and logic in the JSP. 

One thing I am investigating is if the groups and categories could be populated with just the names instead of JSON objects. I did not want to remove groups (as JSON objects) in case they were used somewhere else in the code. Looking at this simplification now.